irispy/spectrograph.py

```diff
@@ -208,6 +209,363 @@ def spectral_windows(self):
+                new_unit = new_data_quantities[0].unit
+        else:
+            raise ValueError("Input unit type not recognized.")
+        return IRISSpectrogram(new_data, self.wcs, new_uncertainty, new_unit, self.meta,
+                               _extra_coords_to_input_format(self._extra_coords, self.missing_axis),
+                               mask=self.mask, missing_axis=self.missing_axis)
+
+    def apply_exposure_time_correction(self, undo=False):
+        """
+        Applies or undoes exposure time correction to data and uncertainty and adjusts unit.
+
+        Parameters
+        ----------
+        undo: `bool`
+            If False, exposure time correction is applied.
+            If True, exposure time correction is undone.
+            Default=False
+
+        copy: `bool`
+            If True a new instance with the converted data values is returned.
+            If False, the current instance is overwritten.
+            Default=False
+
+        Returns
+        -------
+        result: `IRISSpectrogram`
+            New IRISSpectrogram in new units.
+
+        """
+        # Get exposure time in seconds and change array's shape so that
+        # it can be broadcast with data and uncertainty arrays.
+        exposure_time_s = self._extra_coords["exposure time"]["value"].to(u.s).value
+        if not self._extra_coords["exposure time"]["value"].isscalar:
+            if len(self.dimensions.shape) == 1:
+                pass
+            elif len(self.dimensions.shape) == 2:
+                exposure_time_s = exposure_time_s[:, np.newaxis]
+            elif len(self.dimensions.shape) == 3:
+                exposure_time_s = exposure_time_s[:, np.newaxis, np.newaxis]
+            else:
+                raise ValueError(
+                    "IRISSpectrogram dimensions must be 2 or 3. Dimensions={0}".format(
+                        len(self.dimensions.shape)))
+        # Based on value on undo kwarg, apply or remove exposure time correction.
+        if undo is True:
+            new_data_arrays, new_unit = iris_tools.uncalculate_exposure_time_correction(
+                (self.data, self.uncertainty.array), self.unit, exposure_time_s)
+        else:
+            new_data_arrays, new_unit = iris_tools.calculate_exposure_time_correction(
+                (self.data, self.uncertainty.array), self.unit, exposure_time_s)
+        # Return new instance of IRISSpectrogram with correction applied/undone.
+        return IRISSpectrogram(
+            new_data_arrays[0], self.wcs, new_data_arrays[1], new_unit, self.meta,
+            _extra_coords_to_input_format(self._extra_coords, self.missing_axis),
+            mask=self.mask, missing_axis=self.missing_axis)
+
+def _get_meta_values(keys, meta, unknown_message="Unknown"):
+    values = []
+    for key in keys:
+        try:
+            value = meta[key]
+        except KeyError:
+            value = unknown_message
```

Comment:
if `meta` is a  dictionary like, you don't need this `try/except` block but only:
```python
value = meta.get(key, unknown_message)
```

If you want to get a list with all these values in `keys`:
```python
values = [meta.get(key, unknown_message) for key in keys]
```